src/TransitionWrapper/TransitionWrapper.tsx

```diff
@@ -0,0 +1,154 @@
+import * as React from 'react';
+import cx from '../utils/classnames';
+import omit = require('lodash/omit');
+import { props, t, ReactChildren } from '../utils';
+
+export namespace TransitionWrapperProps {
+  export type TransitionStyles = {
+    enter?: React.CSSProperties,
+    enterActive?: React.CSSProperties,
+    default?: React.CSSProperties,
+    leave?: React.CSSProperties,
+    leaveActive?: React.CSSProperties
+  }
+}
+
+export type TransitionWrapperDefaultProps<CP> = {
+  /** object with inline-style for each transition event. It's also possible to use `css` classes (formatted in kebab-case) */
+  transitionStyles: TransitionWrapperProps.TransitionStyles,
+  /** custom component to be used as wrapper for `children`. Can be either an html tag name string (eg. 'div', 'span', etc), or a `ReactClass` (eg. `FlexView`) */
+  component:  keyof React.ReactHTML | React.ComponentClass<CP>,
+  /** callback for componentDidLeave: useful if you need to do some cleanup */
+  onLeave: () => void,
+  style: React.CSSProperties,
+};
+
+export type TransitionWrapperRequiredProps = {
+  /** the component you want to animate (it must have a unique "key") */
+  children: JSX.Element,
+  /** duration of enter transition in milliseconds */
+  transitionEnterTimeout: number,
+  /** duration of leave transition in milliseconds */
+  transitionLeaveTimeout: number,
+  className?: string,
+};
+
+export type TransitionWrapperProps<CP> = TransitionWrapperRequiredProps & Partial<TransitionWrapperDefaultProps<CP>> & {[k in keyof CP]: CP[k]};
```

Comment:
sostituendo {[k in keyof CP]: CP[k]} con un più ovvio CP, abbiamo errori del typechecker, che non riesce a inferire i tipi delle props